File: Data Structures & Algorithms/validate-parentheses/submission-3.cpp
class Solution {
public:
    char inverse(char c){
        if(c == '}') return '{';
        if(c == ']') return '[';
        if(c == ')') return '(';
        //return c;
    }

    bool isValid(string s) {
        stack<char> st;
        for(int i = 0; i< s.size(); i++){
            if(s[i] == '[' || s[i] == '{' || s[i] == '(' ){
                st.push(s[i]);
            }
            else{
                if(!st.empty() && inverse(s[i])== st.top()){
                    st.pop();
                }
                else{
                    return false;
                } 
            }
        }
        if(st.empty()) return true;
        return false;
    }
};

File: Data Structures & Algorithms/daily-temperatures/submission-3.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<int[]> st = new Stack<>();
        int[] res = new int[temperatures.length];
        for (int i = 0; i < temperatures.length; i++) {
            while(!st.isEmpty() && st.peek()[1] < temperatures[i]) {
                int[] a = st.pop();
                res[a[0]] = i - a[0];
            }
            st.push(new int[] {i,temperatures[i]});
        }
        return res;
    }
}
